src/solvers/dgsem_p4est/containers_2d_manifold_in_3d_covariant.jl

```diff
@@ -0,0 +1,241 @@
+@muladd begin
+#! format: noindent
+
+@doc raw"""
+    P4estElementContainerCovariant{NDIMS, RealT <: Real, uEltype <: Real,
+                                   NDIMSP1, NDIMSP2,
+                                   NDIMSP3} <: Trixi.AbstractContainer
+
+Specialized element container for equations in covariant form on a manifold of dimension
+`NDIMS`, created when `Trixi.init_elements` is dispatched on 
+`AbstractCovariantEquations{NDIMS, NDIMS_AMBIENT}` and `P4estMesh{NDIMS, NDIMS_AMBIENT`.
+Contains the following geometric information:
+- `node_coordinates::Array{RealT, NDIMSP2}:` Cartesian components of the node positions
+  within the ambient space of dimension `NDIMS_AMBIENT`, where the first index is the 
+  Cartesian component index, the next `NDIMS` indices are tensor-product node indices ($i$, 
+  $j$, for `NDIMS = 2` or $i$, $j$, $k$, for `NDIMS = 3`), and the last is the element
+  index.
+- `covariant_basis::Array{RealT, NDIMSP3}`: Components of the covariant tangent basis
+  vectors  $\vec{a}_m = \partial \vec{X} / \partial \xi^m$ expanded in terms of a global  
+  tangent basis (i.e. zonal and meridional components in the case of a spherical shell), 
+  where $\vec{X}$ denotes the mapping from the reference element to the physical element.
+  he first index is the global component index $l$, the second index is the local
```

Comment:
```suggestion
  The first index is the global component index $l$, the second index is the local
```

Reply:
Fixed in [4f1b3dc](https://github.com/trixi-framework/TrixiAtmo.jl/pull/31/commits/4f1b3dcea05349139e32a19d5d4ad714ba3b499e).